src/catalog.ts

```diff
@@ -63,25 +64,29 @@ export class Catalog {
     return new Catalog(await promise);
   }
 
-  public getByIdentifier(identifier: string): Dataset | undefined {
-    return this.datasets.find(dataset => dataset.identifier === identifier);
+  public getByDistributionIri(iri: IRI): Dataset | undefined {
```

Comment:
Looks good! One question, though: will this method suffice for usage in package `network-of-terms-comunica`? The method returns a Dataset, but `network-of-terms-comunica` wants a distribution (notably: its endpoint URL).

Reply:
Good point. While this would work, it’s somewhat cumbersome because it forces dependants to retrieve the dataset with all its distributions and _then_ look up the distributions inside that.

However, we need more than just the distribution: the dataset is also [used in building `QueryResult`](https://github.com/netwerk-digitaal-erfgoed/network-of-terms-comunica/blob/9f35a2aeb4449be8cc5796f65c8057df72b3f840/src/services/query.ts#L34) for returning the dataset and IRI along with the search results.

Reply:
> Should we solve this by returning the dataset IRI along with the distribution and add a second lookup method, e.g. Catalog::getDatasetByIri()?

Or we could rename the current lookup method and add a second lookup to the `Dataset` class? For example:

```javascript
class Catalog {
    getDatasetByDistributionIri(iri): Dataset
}

class Dataset {
    getDistributionByIri(iri): Distribution
}
```

Implementation (e.g. in `network-of-terms-comunica`):

```javascript
const distributionIri = 'http://example.org';
const dataset = catalog.getDatasetByDistributionIri(distributionIri);
const distribution = dataset.getDistributionByIri(distributionIri);
```

This feels a bit redundant (two calls with `distributionIri` as parameter), but it keeps the lookup logic inside the `catalog` package.

Reply:
> We may also have to rethink the semantics of listSources(). Should that return a list of datasets (with all their distributions)? Or should it return a list of distribution, each with the dataset it distributes? In other words, is a source still a dataset (as we determined previously) or has it become a distribution (since the primary entry point is now the distribution)?

Good one! Yes, I think we should change `listSources()`. From a user's (e.g. GraphQL) perspective, it should now return a list of distributions. In other words: a `source` is a distribution, something that can be queried. Perhaps we can further discuss - in `network-of-terms-comunica`- if and how this impacts the GraphQL interface (e.g. do we want to expose dataset information there as well or merely distribution information?).

Ideally each distribution should have a reference to the dataset to which it belongs. Although I'm not sure whether this is necessary right now: for identifying and querying a distribution its IRI and name are required - we don't need information from its dataset?

By the way, a distribution currently doesn't (need to) have a `name` property. Should we enforce this? Or should we fallback to the name of the dataset if the name of the distribution is missing?